extensions/interactions/CodeRepl/CodeRepl.js

```diff
@@ -146,3 +146,156 @@ oppia.directive('oppiaShortResponseCodeRepl', [
+       */
+      // TODO(sll): Augment this function to strip out comments that occur at the
+      // end of a line. However, be careful with lines where '#' is contained in
+      // quotes or the character is escaped.
+      var FOUR_SPACES = '    ';
+      // Maps the number of spaces at the beginning of a line to an int specifying
+      // the desired indentation level.
+      var numSpaceToDesiredIndentLevel = {
+          0: 0,
+      };
+
+      var codeLines = removeTrailingWhitespace(codeString).split('\n');
+      var normalizedCodeLines = [];
+      codeLines.forEach(function(line) {
+        if (removeLeadingWhitespace(line).startsWith('#')) {
+          return;
+        }
+        line = removeTrailingWhitespace(line);
+        if (!line) {
+          return;
+        }
+
+        var numSpaces = line.length - removeLeadingWhitespace(line).length;
+
+        var maxNumSpaces = Object.keys(numSpaceToDesiredIndentLevel).reduce(
+          function(max, key) {
+            return Math.max(key, max);
+          }, -Infinity);
+        if (numSpaces > maxNumSpaces) {
+          // Add a new indentation level
+          numSpaceToDesiredIndentLevel[numSpaces] =
+            Object.keys(numSpaceToDesiredIndentLevel).length;
+        }
+
+        // This is set when the indentation level of the current line does not
+        // start a new scope, and also does not match any previous indentation
+        // level. This case is actually invalid, but for now, we take the
+        // largest indentation level that is less than this one.
+        // TODO(sll): Bad indentation should result in an error nearer the
+        // source.
+        var isShortfallLine =
+          !numSpaceToDesiredIndentLevel.hasOwnProperty(numSpaces) &&
+          numSpaces < maxNumSpaces;
+
+        // Clear all existing indentation levels to the right of this one.
+        for (var key in numSpaceToDesiredIndentLevel) {
+          if (key > numSpaces) {
+            delete numSpaceToDesiredIndentLevel[key];
+          }
+        }
+
+        if (isShortfallLine) {
+          numSpaces = Object.keys(numSpaceToDesiredIndentLevel).reduce(
+            function(max, key) {
+              return Math.max(numSpaceToDesiredIndentLevel[key], max);
+            }, -Infinity);
+        }
+
+        var normalizedLine = '';
+        for (var i = 0; i < numSpaceToDesiredIndentLevel[numSpaces]; i++) {
+          normalizedLine += FOUR_SPACES;
+        }
+        normalizedLine += removeLeadingWhitespace(line);
+        normalizedCodeLines.push(normalizedLine);
+      });
+      return normalizedCodeLines.join('\n');
+    },
+    // A very naive approach to 'normalizing' the code is to strip out all
+    // comments and whitespace. This normalization currently assumes Python.
+    normalizePythonCode: function(code) {
+      // TODO(sll): This does not correctly handle the case where '#' is
+      // within quotes, or where it is escaped.
+      // remove comments
```

Comment:
Please capitalize this comment, and put a full stop at the end; ditto two lines below.